### PR TITLE
Span device qualifier

### PR DESCRIPTION
**Pull request by fweik**
and remove `CXX14_CONSTEXPR` macro

### CODE REVIEW

File: src/core/utils/device_qualifier.hpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+#define UTILS_DEVICE_QUALIFIER_HPP
+
+#if defined(__CUDACC__)
+#define DEVICE_QUALIFIER __host__ __device__
+#else
+#define DEVICE_QUALIFIER
+#endif
+
+#ifdef __CUDACC__
+#define DEVICE_ASSERT(A) assert((A))
```
Comment (KaiSzuttor) [inline src/core/utils/device_qualifier.hpp:9]: Why not have all defines in one ```ifdef``` ?

Comment (fweik) [reply src/core/utils/device_qualifier.hpp:9]: Not the same logic

Comment (KaiSzuttor) [reply src/core/utils/device_qualifier.hpp:9]: it's binary logic that depends on ```__CUDACC__```, and both cases are handled at the top. What do I miss here?

Comment (fweik) [reply src/core/utils/device_qualifier.hpp:9]: one is `ifndef`

**Review by KaiSzuttor — COMMENTED** [review]
good that you also think it's important :smile:

**Comment by fweik** [comment]
@KaiSzuttor

**Review by KaiSzuttor — APPROVED** [review]

**Pull request merged**